platform/source/TARGET_CORTEX_M/TOOLCHAIN_IAR/except.S

```diff
@@ -42,114 +42,75 @@ FAULT_TYPE_USAGE_FAULT          EQU     0x40
 
-        MRS     R0,MSP
-        LDR     R1,=0x4
-        MOV     R2,LR
-        TST     R2,R1                   ; Check EXC_RETURN for bit 2
-        BEQ     Fault_Handler_Continue
-        MRS     R0,PSP
+        MOV     R12,R3
+        PUSH    {R4-R7}
+        MOV     R6,SP
+        MOV     R5,LR
+        LSRS    R0,R5,#2                ; Check EXC_RETURN for bit 2
+        BCC     Fault_Handler_Continue
+        MRS     R6,PSP
 
 Fault_Handler_Continue
-        MOV     R12,R3
-        LDR     R3,=mbed_fault_context
-        LDR     R1,[R3]
-        LDR     R2,[R0]                 ; Capture R0
-        STR     R2,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#4]              ; Capture R1
-        STR     R2,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#8]              ; Capture R2
-        STR     R2,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#12]             ; Capture R3
-        STR     R2,[R1]
-        ADDS    R1,#4
-        STMIA   R1!,{R4-R7}             ; Capture R4..R7
-        MOV     R7,R8                   ; Capture R8
-        STR     R7,[R1]
-        ADDS    R1,#4
-        MOV     R7,R9                   ; Capture R9
-        STR     R7,[R1]
-        ADDS    R1,#4
-        MOV     R7,R10                  ; Capture R10
-        STR     R7,[R1]
-        ADDS    R1,#4
-        MOV     R7,R11                  ; Capture R11
-        STR     R7,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#16]             ; Capture R12
-        STR     R2,[R1]
-        ADDS    R1,#8                   ; Add 8 here to capture LR next, we will capture SP later
-        LDR     R2,[R0,#20]             ; Capture LR
-        STR     R2,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#24]             ; Capture PC
-        STR     R2,[R1]
-        ADDS    R1,#4
-        LDR     R2,[R0,#28]             ; Capture xPSR
-        STR     R2,[R1]
-        ADDS    R1,#4
-        ; Adjust stack pointer to its original value and capture it
-        MOV     R3,R0
-        ADDS    R3,#0x20                ; Add 0x20 to get the SP value prior to exception
-        LDR     R6,=0x200
-        TST     R2,R6                   ; Check for if STK was aligned by checking bit-9 in xPSR value
-        BEQ     Fault_Handler_Continue1
-        ADDS    R3,#0x4
+        LDR     R7,=mbed_fault_context
+        LDR     R7,[R7]
+        LDMIA   R6!,{R0-R3}
+        STMIA   R7!,{R0-R3}             ; Capture R0..R3
+        POP     {R0-R3}
+        STMIA   R7!,{R0-R3}             ; Capture R4..R7
+        MOV     R0,R8
+        MOV     R1,R9
+        MOV     R2,R10
+        MOV     R3,R11
+        STMIA   R7!,{R0-R3}             ; Capture R8..R11
+        LDMIA   R6!,{R0,R2-R4}          ; Load R12,LR,PC,xPSR
+        ; Adjust stack pointer to its original value
+        MOVS    R1,R6
+        LSRS    R6,R4,#9                ; Check for if STK was aligned by checking bit-9 in xPSR value
+        BCC     Fault_Handler_Continue1
+        ADDS    R1,#0x4
```

Comment:
Also here: can we comment on this addition or its me who does not see it before going into all instructions prior this one?

Reply:
That's performing the 8-byte alignment referenced (or at least hinted at) in the comment above.